test/unit/sthDatabase_test.js

```diff
@@ -118,117 +126,164 @@ function connectToDatabase(callback) {
-        // The collection does not exist
+    sthConfig.DATA_MODEL = dataModel;
+    var collectionName =
+        dataType === sthTestConfig.DATA_TYPES.RAW
+            ? sthDatabaseNaming.getRawCollectionName(collectionNameParams)
+            : sthDatabaseNaming.getAggregatedCollectionName(collectionNameParams);
+
+    if (collectionName) {
+        sthDatabase.connection.dropCollection(collectionName, function(err) {
+            if (err && err.code === 26 && err.name === 'MongoError' && err.message === 'ns not found') {
+                // The collection does not exist
+                return process.nextTick(callback);
+            }
+            return process.nextTick(callback.bind(null, err));
+        });
+    } else {
         return process.nextTick(callback);
-      }
-      return process.nextTick(callback.bind(null, err));
-    });
-  } else {
-    return process.nextTick(callback);
-  }
+    }
 }
 
 /**
  * Set of tests to drop the test collections from the database
  */
 function cleanDatabaseTests() {
-  var dataModelsKeys = Object.keys(sthConfig.DATA_MODELS),
-      dataTypeKeys = Object.keys(sthTestConfig.DATA_TYPES);
-
-  describe('database clean up', function() {
-    dataModelsKeys.forEach(function(dataModel) {
-      dataTypeKeys.forEach(function(dataType) {
-        it('should drop the ' + sthConfig.DATA_MODELS[dataModel] + ' ' + sthTestConfig.DATA_TYPES[dataType] +
-          ' data collection if it exists',
-          dropCollection.bind(null, COLLECTION_NAME_PARAMS, sthTestConfig.DATA_TYPES[dataType],
-            sthConfig.DATA_MODELS[dataModel]));
-
-        it('should drop the numeric ' + sthConfig.DATA_MODELS[dataModel] + ' ' + sthTestConfig.DATA_TYPES[dataType] +
-          ' data collection if it exists',
-          dropCollection.bind(null, NUMERIC_COLLECTION_NAME_PARAMS, sthTestConfig.DATA_TYPES[dataType],
-            sthConfig.DATA_MODELS[dataModel]));
-
-        it('should drop the textual ' + sthConfig.DATA_MODELS[dataModel] + ' ' + sthTestConfig.DATA_TYPES[dataType] +
-          ' data collection if it exists',
-          dropCollection.bind(null, TEXTUAL_COLLECTION_NAME_PARAMS, sthTestConfig.DATA_TYPES[dataType],
-            sthConfig.DATA_MODELS[dataModel]));
-
-        it('should drop the very long service path ' + sthConfig.DATA_MODELS[dataModel] + ' ' +
-          sthTestConfig.DATA_TYPES[dataType] + ' data collection if it exists',
-          dropCollection.bind(null, VERY_LONG_COLLECTION_NAME_PARAMS, sthTestConfig.DATA_TYPES[dataType],
-            sthConfig.DATA_MODELS[dataModel]));
-      });
+    var dataModelsKeys = Object.keys(sthConfig.DATA_MODELS),
+        dataTypeKeys = Object.keys(sthTestConfig.DATA_TYPES);
+
+    describe('database clean up', function() {
+        dataModelsKeys.forEach(function(dataModel) {
+            dataTypeKeys.forEach(function(dataType) {
+                it(
+                    'should drop the ' +
```

Comment:
Excesive line break here and other similar places in this file. Please do a general review of it.

Reply:
Fixed  c00464eb